src/pkg/packager/composer/list.go

```diff
@@ -0,0 +1,173 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package composer contains functions for composing components within Zarf packages.
+package composer
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/defenseunicorns/zarf/src/pkg/layout"
+	"github.com/defenseunicorns/zarf/src/pkg/oci"
+	"github.com/defenseunicorns/zarf/src/pkg/utils"
+	"github.com/defenseunicorns/zarf/src/pkg/utils/helpers"
+	"github.com/defenseunicorns/zarf/src/types"
+)
+
+// Node is a node in the import chain
+type Node struct {
+	types.ZarfComponent
+
+	prev *Node
+	next *Node
+}
+
+// ImportChain is a doubly linked list of components
+type ImportChain struct {
+	head *Node
+	tail *Node
+}
+
+func (ic *ImportChain) append(c types.ZarfComponent) {
+	node := &Node{ZarfComponent: c, prev: nil, next: nil}
+	if ic.head == nil {
+		ic.head = node
+		ic.tail = node
+	} else {
+		p := ic.head
+		for p.next != nil {
+			p = p.next
+		}
```

Comment:
You should be able to do this based on the tail pointer instead of traversing forward through the list like this.